tests/e2e/state.go

```diff
@@ -1,52 +1,72 @@
 package main
 
 import (
-	"bufio"
-	"encoding/json"
 	"fmt"
 	"log"
-	"os/exec"
 	"regexp"
-	"sort"
-	"strconv"
-	"strings"
 	"time"
 
 	clienttypes "github.com/cosmos/ibc-go/v8/modules/core/02-client/types"
-	"github.com/kylelemons/godebug/pretty"
-	"github.com/tidwall/gjson"
-	"gopkg.in/yaml.v2"
 
 	e2e "github.com/cosmos/interchain-security/v6/tests/e2e/testlib"
-	"github.com/cosmos/interchain-security/v6/x/ccv/provider/types"
 )
 
 // type aliases
 type (
 	ChainState                   = e2e.ChainState
-	Proposal                     = e2e.Proposal
-	Rewards                      = e2e.Rewards
-	TextProposal                 = e2e.TextProposal
-	UpgradeProposal              = e2e.UpgradeProposal
 	ConsumerAdditionProposal     = e2e.ConsumerAdditionProposal
 	ConsumerRemovalProposal      = e2e.ConsumerRemovalProposal
 	ConsumerModificationProposal = e2e.ConsumerModificationProposal
 	IBCTransferParams            = e2e.IBCTransferParams
 	IBCTransferParamsProposal    = e2e.IBCTransferParamsProposal
 	Param                        = e2e.Param
 	ParamsProposal               = e2e.ParamsProposal
+	Proposal                     = e2e.Proposal
+	Rewards                      = e2e.Rewards
 	TargetDriver                 = e2e.TargetDriver
+	TextProposal                 = e2e.TextProposal
+	TxResponse                   = e2e.TxResponse
+	UpgradeProposal              = e2e.UpgradeProposal
 )
 
 type State map[ChainID]ChainState
 
 type Chain struct {
-	target     e2e.TargetDriver
+	target     TargetDriver
 	testConfig *TestConfig
 }
 
+// waitForTx waits until a transaction is seen in a block or panics if a timeout occurs
+func (tr Chain) waitForTx(chain ChainID, txResponse []byte, timeout time.Duration) TxResponse {
+	// remove any gas estimate as when command is run with gas=auto the output contains gas estimation mixed with json output
+	re, err := regexp.Compile("gas estimate: [0-9]+")
+	if err != nil {
+		panic(fmt.Sprintf("error compiling regexp: %s", err.Error()))
+	}
+	txResponse = re.ReplaceAll(txResponse, []byte{})
+
+	// check transaction
+	response := e2e.GetTxResponse(txResponse)
+	if response.Code != 0 {
+		log.Fatalf("sending transaction failed with error code %d, Log:'%s'", response.Code, response.RawLog)
+	}
+
+	// wait for the transaction
+	start := time.Now()
+	for {
+		res, err := tr.target.QueryTransaction(chain, response.TxHash)
```

Comment:
maybe we can add some sleep of one second(or commit time), just to avoid querying too often